src/MoBi.Presentation/Presenter/ChartPresenter.cs

```diff
@@ -191,15 +193,36 @@ protected virtual void OnDragDrop(object sender, IDragEvent e)
          }
          else
          {
-            var droppedObservedData = _observedDataDragDropBinder.DroppedObservedDataFrom(e);
-            addObservedData(droppedObservedData.ToList());
+            if (_userSettings.ColorGroupObservedDataFromSameFolder)
+            {
+               var droppedObservedDataWithFolderAddress = _observedDataDragDropBinder.DroppedObservedDataWithFolderPathFrom(e);
```

Comment:
@msevestre so to my understanding this has to do with userSettings, that actually exists in every project seperately, since they are not exactly implemented the same way. That's why the duplication. If you think the duplication should be removed, let's discuss this, since I am not 100% sure about the best way to implement this ( base function that also gets ICoreUserSettings as an argument? not sure this would work since ColorGroupObservedDataFromSameFolder exists in IPresentationUserSettings )

Reply:
> that actually exists in every project seperately, since they are not exactly implemented the same way

I don't understand. Why isn't it implemented in the same way? Let's chat